_ctoApps/caesar/src/caesar/fragment.html

```diff
@@ -114,21 +121,189 @@
+            <button type="button" class="close" data-dismiss="modal">&times;</button>
+        </div>
+        <div class="modal-body">
+            <p class="mb-0">${{caesar.RESET_CODE_MODAL_CONTENT}}$</p>
+        </div>
+        <div class="modal-footer">
+            <button id="caesar-python-local-reset-code" class="btn btn-primary"><i class="fa fa-check mr-1"></i>${{caesar.YES}}$</button>
+            <button class="btn btn-secondary" data-dismiss="modal"><i class="fa fa-times mr-1"></i>${{caesar.NO}}$</button>
+        </div>
+    </div></div></div>
+
+</div>
+
+<div id="caesar-python-remote-collapse" class="collapse">
+    <div class="alert alert-info mb-2 mt-3">${{caesar.PYTHON_REMOTE_NOTICE}}$</div>
+    <iframe height="900px" width="100%" scrolling="no" frameborder="no" allowtransparency="true" allowfullscreen="true" sandbox="allow-forms allow-pointer-lock allow-popups allow-same-origin allow-scripts allow-modals"></iframe> <!-- src will be set via JS -->
+</div>
+
+<script>
+
+    // toggle local collapse on button click
+    $("#caesar-run-as-python-local").click(function() {
+        $("#caesar-python-local-collapse").collapse("toggle")
+        $("#caesar-python-remote-collapse").collapse("hide")
+    })
+
+    let localPythonAssetsLoaded = false
+
+    // when local collapse gets shown -> set button state and include Python editor (also use Python for calculations)
+    $("#caesar-python-local-collapse").on("show.bs.collapse", function() {
+        $("#caesar-run-as-python-local").html('<i class="fa fa-chevron-up mr-1"></i> ${{caesar.HIDE_LOCAL_PYTHON}}$')
+
+        // include assets for local Python editor
+        if(!localPythonAssetsLoaded) {
+            let localPythonStyles = [
+                "resources/codemirror/lib/codemirror.css",
+                "resources/codemirror/theme/shadowfox.css",
+                "resources/codemirror/theme/eclipse.css",
+                "resources/codemirror/theme/yeti.css",
+                "resources/codemirror/theme/solarized.css"
+            ]
+            localPythonStyles.forEach(styleURL => {
+                let styleElem = document.createElement("link")
+                styleElem.type = "text/css"; styleElem.rel = "stylesheet"
+                styleElem.href = CTO_Globals.pluginRoot + styleURL
+                document.head.appendChild(styleElem)
+            })
+            let localPythonScripts = [
+                "resources/codemirror/lib/codemirror.js",
+                "resources/codemirror/mode/python/python.js",
+                "resources/pyodide.js", // maybe load via CDN?
+                "caesar-python-" + window.ioApp.lang + ".js"
+            ]
+            localPythonScripts.forEach(scriptURL => {
+                let scriptElem = document.createElement("script")
+                scriptElem.src = CTO_Globals.pluginRoot + scriptURL
+                $("#caesar-python-local-scripts").append(scriptElem)
+            })
+            localPythonAssetsLoaded = true
```

Comment:
Not sure this is a good assumption to consider that the scripts are loaded after this point. Maybe consider using the `.addEventListener("load"` method on each `scriptElem` to wait until they're loaded.

Reply:
This variable would better be called something like `localPythonAssetsInserted` as it's just an indicator for if the assets have been injected already, so that they don't get inserted again when the collapse opens again. But the `load` event could be a part of a solution for loading the Pyodide script from its CDN.

Reply:
The thing is that on your locale machine it works because files are loading instantly. Even if we don't use a CDN, scripts will take some time to load on cryptool.org. So this needs to be fixed before we merge 😄 !